plugins/woocommerce/includes/admin/settings/views/html-admin-page-shipping-zone-methods.php

```diff
@@ -198,9 +198,27 @@
 								if ( ! $method->supports( 'shipping-zones' ) ) {
 									continue;
 								}
-								$description = wp_kses_post( $method->get_method_description() );
-								echo '<div class="wc-shipping-zone-method-input"><input type="radio" value="' . esc_attr( $method->id ) . '" id="' . esc_attr( $method->id ) . '" name="add_method_id"/><label for="' . esc_attr( $method->id ) . '">' . esc_html( $method->get_method_title() ) . '<span class="dashicons dashicons-yes"></span></label><div class="wc-shipping-zone-method-input-help-text"><span>' . esc_html( $description ) . '</span></div></div>';
+
+								echo '<div class="wc-shipping-zone-method-input"><input type="radio" value="' . esc_attr( $method->id ) . '" id="' . esc_attr( $method->id ) . '" name="add_method_id"/><label for="' . esc_attr( $method->id ) . '">' . esc_html( $method->get_method_title() ) . '<span class="dashicons dashicons-yes"></span></label></div>';
+							}
+
+							echo '<div class="wc-shipping-zone-method-input-help-text-container">';
+
+							foreach ( $methods_placed_in_order as $method ) {
+								if ( ! $method->supports( 'shipping-zones' ) ) {
+									continue;
+								}
+								$allowed_html = array(
+									'a' => array(
+										'href'   => true,
+										'title'  => true,
+										'target' => true,
+									),
+								);
+
+								echo '<div id=' . esc_attr( $method->id ) . '-description class="wc-shipping-zone-method-input-help-text"><span>' . wp_kses( $method->get_method_description(), $allowed_html ) . '</span></div>';
```

Comment:
I looked at the previous shipping implementation, it looks like it used [`wp_kses_post`](https://github.com/woocommerce/woocommerce/blob/8a3a9bc356ab7c1820921250b0425dd689b0d513/plugins/woocommerce/includes/admin/settings/views/html-admin-page-shipping-zones.php#L114) which is a shorthand for allowing [these attributes](https://github.com/WordPress/wordpress-develop/blob/f90515acdb5313e14445868e25b2b19acaa58ee7/src/wp-includes/kses.php#L68C27-L68C27) plus other stuff. Though it does not explicitly allow `a`'s `title` like yours.

My thought is if we want to maximize backwards compatibility, we could lean towards reusing `wp_kses_post`. However, I'm not familiar with how 3PD uses `get_method_description` and the previous method might not be suitable with our new approach. What do you think, @psealock?

Reply:
Thats a great call, thank you @ilyasfoo. Looking at the previous implementation, [wpautop](https://developer.wordpress.org/reference/functions/wpautop/) was also used for formatting purposes. So added that back in as well as used `wp_kses_post`.